core/server/master/src/test/java/alluxio/master/metrics/MetricsStoreTest.java

```diff
@@ -66,4 +66,18 @@ public void putClientMetrics() {
         Metric.from("client.192_1_1_1:B.metric1", 15, MetricType.GAUGE)),
         mMetricStore.getMetricsByInstanceTypeAndName(MetricsSystem.InstanceType.CLIENT, "metric1"));
   }
+
+  @Test
+  public void putTaggedMetrics() {
+    String name = "test";
+    Metric metric1 =
+        new Metric(MetricsSystem.InstanceType.WORKER, "host", MetricType.COUNTER, name, 1.0);
+    metric1.addTag("Tag", "1");
+    Metric metric2 =
+        new Metric(MetricsSystem.InstanceType.WORKER, "host", MetricType.COUNTER, name, 2.0);
+    metric2.addTag("Tag", "2");
+    mMetricStore.putWorkerMetrics("host", Lists.newArrayList(metric1, metric2));
+    assertEquals(mMetricStore.getMetricsByInstanceTypeAndName(MetricsSystem.InstanceType.WORKER,
+        name), Sets.newHashSet(metric1, metric2));
+  }
```

Comment:
This test specifically tests that the added metrics contain the tags, but in light of fixing #9442, I think it would also be appropriate to write a new or test add assertions in an already existing test which makes sure UFS read/write values are updated.

Reply:
Before we would not consider these two metrics as distinct. In the case of UFS metrics, that often meant we 0-ed them out.

Reply:
Do you think that it isn't appropriate to add a test that verifies that UFS metrics are updated?

Reply:
I think we should add the test, I was clarifying what this test validates.